python/tank/authentication/sso_saml2/core/sso_saml2_core.py

```diff
@@ -85,6 +85,35 @@ def __init__(self, window_title="Web Login", qt_modules=None):
         if QtWebKit is None:
             raise SsoSaml2MissingQtWebKit("The QtWebKit module is unavailable")
 
+        class TKWebPage(QtWebKit.QWebPage):
+            """
+            Wrapper class to better control the behaviour when clicking on links
+            in the Qt web browser. If we are asked to open a new tab/window, then
+            we defer the page to the external browser.
+
+            We need to open some links in an external window so as to avoid
+            breaking the authentication flow just to visit an external link.
+            Some examples of links that the user may see which we want to open
+            externally:
+             - Term of use and conditions,
+             - Download of the Google/Duo authenticator app
+             - Any other links which may be presented by SSO Providers
+            """
+
+            def acceptNavigationRequest(self, frame, request, n_type): # noqa
+                """
+                Overloaded method, to properly control the behavioir of clicking on
+                links.
```

Comment:
[nitpick] Would be nice to see param docs

Reply:
Will add the docs.
Since this was a callback for Qt itself, and not something we could call ourselves, I figured that this was not required... But it is good to have it when debugging/upgrading to QtX+1

Reply:
Yeah, the standard in Toolkit is to always document methods, whether they are public or not and always using sphinx compatible syntax.